include/dsn/internal/rpc_message.h

```diff
@@ -63,17 +63,52 @@ namespace dsn
                              // we leverage for optimization (fast rpc handler lookup)
     };
 
+    struct header_type
+    {
+    public:
+        union
+        {
+            char stype[4];
+            int32_t itype;
+        } type;
+        header_type()
+        {
+            type.itype = -1;
+        }
+        header_type(const char* str)
+        {
+            memcpy(type.stype, str, sizeof(int32_t));
+        }
+        header_type(const header_type& another)
+        {
+            type.itype = another.type.itype;
+        }
+        header_type& operator=(const header_type& another)
+        {
+            type.itype = another.type.itype;
+            return *this;
+        }
+        bool operator==(const header_type& other) const
+        {
+            return type.itype==other.type.itype;
+        }
+    public:
+        static header_type hdr_dsn_default;
+        static header_type hdr_dsn_thrift;
+    };
+
     typedef struct message_header
     {
-        int32_t        hdr_crc32;
-        int32_t        body_crc32;
-        int32_t        body_length;
+        header_type    hdr_type;
+        uint32_t       hdr_crc32;
+        uint32_t       body_crc32;
+        uint32_t       body_length;
         int32_t        version;
```

Comment:
ok, I'll add a field named 'hdr_length' to meet the possibility of extend message header.
then:
        header_type    hdr_type;
        uint32_t       hdr_version;
        uint32_t       hdr_length;
        uint32_t       hdr_crc32;
        uint32_t       body_length;
        uint32_t       body_crc32;